contracts/Minter.sol

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts (last updated v4.5.0) (token/ERC20/presets/ERC20PresetMinterPauser.sol)
+
+pragma solidity ^0.8.7;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "./samples/AuthorizedAddresses.sol";
+
+contract Minter is ERC20, AuthorizedAddresses {
```

Comment:
Shouldn't we rename the contract from `Minter`to `Factory`? (and thus update all the `Minter` references in the project).